sdk/src/main/java/velocitas/sdk/Logger.kt

```diff
@@ -0,0 +1,115 @@
+package velocitas.sdk
+
+import java.util.Date
+
+interface ILogger {
+    /**
+     * Log a message with info level.
+     *
+     * @param msg The message to log.
+     */
+    fun info(msg: String)
+
+    /**
+     * Log a message with warn level.
+     *
+     * @param msg The message to log.
+     */
+    fun warn(msg: String)
+
+    /**
+     * Log a message with error level.
+     *
+     * @param msg The message to log.
+     */
+    fun error(msg: String)
+
+    /**
+     * Log a message with debug level.
+     *
+     * @param msg The message to log.
+     */
+    fun debug(msg: String)
+}
+
+object Logger {
+    private var impl: ILogger = ConsoleLogger()
```

Comment:
Well, let's call it an uncommon strategy pattern. I don't think the abstraction is bad per-say. You can now easily also implement a "FileLogger" or other stuff, because it's quite flexible.